packages/dropzone/src/core.ts

```diff
@@ -0,0 +1,200 @@
+import attrAccepts from "attr-accept";
+
+import type { AcceptProp, DropzoneState, FileWithPath } from "./types";
+
+// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+const accepts: typeof attrAccepts =
+  // @ts-expect-error - ESM interop
+  typeof attrAccepts === "function" ? attrAccepts : attrAccepts.default;
+
+// Firefox versions prior to 53 return a bogus MIME type for every file drag, so dragovers with
+// that MIME type will always be accepted
+export function isFileAccepted(file: File, accept: string | string[]) {
+  return file.type === "application/x-moz-file" || accepts(file, accept);
+}
+
+export function isEnterOrSpace(event: { key?: string; keyCode?: number }) {
+  return (
+    ("key" in event && (event.key === " " || event.key === "Enter")) ||
+    ("keyCode" in event && (event.keyCode === 32 || event.keyCode === 13))
+  );
+}
+
+const isDefined = <T>(v: T | null | undefined): v is T => v != null;
+export function isValidSize(file: File, minSize: number, maxSize: number) {
+  if (isDefined(file.size)) {
```

Comment:
Can we early escape if not defined maybe? Would clean up the nesting of this logic a bit

Reply:
Oh is this code ripped from the lib? If so might be worth commenting on top

Reply:
Most of it is the same as before (i.e. forked from react-dropzone) and some of it I cleaned up a bit when removing stuff we didnt need